app/main.py

```diff
@@ -2,5 +2,17 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_history = dict()
+
+    def wrapper(*args) -> str:
+        # func_name = f"{func}({args})"
+
+        if f"{func}({args})" in cache_history:
```

Comment:
```suggestion
        if args in cache_history:
```
Why do you need func?
Remove it from all places

Reply:
I added the function name to the cache because I moved the cache to outside the decorator. In this case arguments could be repeated for different functions.

I corrected it as you recommended.
What else can I add to make the code better?